doc/lua_api.txt

```diff
@@ -3345,6 +3345,19 @@ It can be created via `Settings(filename)`.
     * write changes to file
 * `to_table()`: returns `{[key1]=value1,...}`
 
+### `RaycastState`
+A raycast on the map. It works with selection boxes.
+
+It can be created via `RaycastState(pos1, pos2, objects, liquids)` where:
+    * `pos1`: start of the shootline
+    * `pos2`: end of the shootline
+    * `objects` : if false, only nodes will be returned. Default is true.
+    * `liquids' : if false, liquid nodes won't be returned. Defalt is false.
+
+#### Methods
+* `next()`: returns a `pointed_thing`
+    * Returns the next thing pointed by the shootline.
```

Comment:
Instead of (or in addition to) `next()` I think you should use something like `iter()`, so that something like this will work:
```Lua
for x in RaycastState(p1, p2).iter() do
    -- ...
end
```